src/AllensCausalModel/CausalModels.jl

```diff
@@ -0,0 +1,57 @@
+module CausalModels
+using Distributions
+using Statistics
+export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+	#represent as function with id
+	function ExogenousVariable(name,dist)
+		id=rand(1)
+		function d(x;calculated=Dict())
+			if x=="id"
+				return id
+			elseif haskey(calculated,d("id"))
+				return get(calculated,d("id"),0)
+			elseif name in keys(x)
+				return x[name]
+			else
+				return rand(dist,1)[1]
+			end
+		end
+		return d
+	end
+	#represent as function with id
+	function EndogenousVariable(func, args)
+		id=rand(1)
+		function d(x;calculated=Dict())
+			if x=="id"
+				return id
+			elseif haskey(calculated,d("id"))
+				return get(calculated,d("id"),0)
+			end
+			function eval(f)
+				retval=f(x,calculated=calculated)
+				push!(calculated,f("id")=>retval)
```

Comment:
A simpler and more idiomatic way to set a member of a dictionary is `calculated[f("id")] = retval`